### PR TITLE
Allow configuring `DefaultPrettyPrinter` separators for empty Arrays and Objects

### PR DESCRIPTION
**Pull request by gulecroc**
Fix for https://github.com/FasterXML/jackson-core/issues/1128

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java [hunk 1]
```diff
@@ -110,10 +110,20 @@ public interface Indenter
      */
     protected String _objectEntrySeparator;
 
+    /**
+     * @since 2.17
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java:114]: Could you also add a brief explanation of what these do?

Comment (gulecroc) [reply src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java:116]: Hi @cowtowncoder, I fix it

Comment (cowtowncoder) [reply src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java:116]: Thanks!

File: src/main/java/com/fasterxml/jackson/core/util/Separators.java [hunk 2]
```diff
@@ -97,26 +114,30 @@ public Separators(
             Spacing objectFieldValueSpacing,
             char objectEntrySeparator,
             Spacing objectEntrySpacing,
+            String objectEmptySeparator,
             char arrayValueSeparator,
-            Spacing arrayValueSpacing
+            Spacing arrayValueSpacing,
+            String arrayEmptySeparator
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/core/util/Separators.java:120]: This changes signature, no longer same method added in 2.16; I'll change it to chain to new method

**Comment by cowtowncoder** [comment]
@gulecroc Looks good in general: thank you for the changes requested! I will make one small change for backwards compatibility but I think we are good otherwise.

One process thing: if you haven't sent CLA yet (I don't think so but sometimes I forget), we'd need one from here:

https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf

and it's usually easiest to print, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
This is only needed before merging the first contribution; it's good for any future contributions.

Looking forward to merging this nice addition.

**Comment by cowtowncoder** [comment]
Made the compatibility change so all we need is CLA and can merge this in!

**Comment by gulecroc** [comment]
Hi @cowtowncoder, I just sent the CLA.

**Comment by cowtowncoder** [comment]
CLA received, can proceed with merging.

**Review by cowtowncoder — APPROVED** [review]

**Pull request merged**